Assets/Scripts/Tools/LockVertex.cs

```diff
@@ -92,6 +95,28 @@ public void Lock(Vertex currentVertex, bool sendVertexLockEvent = true)
         materialSwap.material = locked;
         currentVertex.GetComponent<MoveVertices>().isLocked = true;
 
+        foreach(Edge e in meshRebuilder.edgeObjects)
```

Comment:
couldn't you go through the vertex's connectedFaces and connectedEdges array instead of all the edges/faces?